### PR TITLE
Fix incorrect property types

### LINKED ISSUE #56349: Incorrect property types
### Issue description

There are differences between docs and property types
```
doc/classes/Polygon2D.xml
74:		<member name="bones" type="Array" setter="_set_bones" getter="_get_bones" default="[]">

scene/2d/polygon_2d.cpp
665:	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "bones", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NO_EDITOR), "_set_bones", "_get_bones");
```

```

doc/classes/RayCast3D.xml
121:		<member name="debug_shape_thickness" type="float" setter="set_debug_shape_thickness" getter="get_debug_shape_thickness" default="2.0">


scene/3d/ray_cast_3d.cpp
355:	ADD_PROPERTY(PropertyInfo(Variant::INT, "debug_shape_thickness", PROPERTY_HINT_RANGE, "1,5"), "set_debug_shape_thickness", "get_debug_shape_thickness");
```

```

scene/3d/voxel_gi.cpp
230:	ADD_PROPERTY(PropertyInfo(Variant::INT, "dynamic_range", PROPERTY_HINT_RANGE, "0,8,0.01"), "set_dynamic_range", "get_dynamic_range");


doc/classes/VoxelGIData.xml
63:		<member name="dynamic_range" type="float" setter="set_dynamic_range" getter="get_dynamic_range" default="4.0">
```
```
scene/resources/visual_shader.cpp
2018:	ADD_PROPERTY(PropertyInfo(Variant::STRING, "engine_version", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NO_EDITOR), "set_engine_version", "get_engine_version");

doc/classes/VisualShader.xml
149:		<member name="engine_version" type="Dictionary" setter="set_engine_version" getter="get_engine_version" default="{}">
```
```
modules/websocket/websocket_server.cpp
70:	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "handshake_timeout"), "set_handshake_timeout", "get_handshake_timeout");

modules/websocket/doc_classes/WebSocketServer.xml
77:		<member name="handshake_timeout" type="float" setter="set_handshake_timeout" getter="get_handshake_timeout" default="3.0">
```
```

doc/classes/XRPose.xml
24:		<member name="has_tracking_data" type="bool" setter="set_has_tracking_data" getter="get_has_tracking_data" default="false">

servers/xr/xr_pose.cpp
38:	ADD_PROPERTY(PropertyInfo(Variant::STRING, "has_tracking_data"), "set_has_tracking_data", "get_has_tracking_data");

```
```

scene/main/viewport.cpp
3953:	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "size_2d_override"), "set_size_2d_override", "get_size_2d_override");

doc/classes/SubViewport.xml
29:		<member name="size_2d_override" type="Vector2i" setter="set_size_2d_override" getter="get_size_2d_override" default="Vector2i(0, 0)">


```

### Steps to reproduce

Not needed

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by binarydigitz01**
Fixes #56349

### CODE REVIEW

**Comment by Chaosus** [comment]
You need to squash all commits as required by our pipeline (see https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request)

**Comment by Ice-Cube69** [comment]
Yeah i've done it now

**Comment by jmb462** [comment]
@Ice-Cube69 After changing the setter and getter for debug_shape_thickness, you need to revert changes at line 355 to : 
https://github.com/godotengine/godot/blob/bc2445d0eb1b4d00de7506fbea4384212c8dfe16/scene/3d/ray_cast_3d.cpp#L355

because now, it's OK to have INT here.

**Comment by Ice-Cube69** [comment]
Reverted and pushed the changes

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -3950,7 +3950,7 @@ void SubViewport::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_clear_mode"), &SubViewport::get_clear_mode);
 
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "size"), "set_size", "get_size");
```
Comment (akien-mga) [inline scene/main/viewport.cpp:3952]: This also seems to be a `Size2i`, i.e. `VECTOR2I`.

**Review by akien-mga — APPROVED** [review]
There's one more to fix, otherwise looks good :+1:

**Comment by akien-mga** [comment]
You should also update `doc/classes/RayCast3D.xml` to propagate the type change from `float` to `int`. You can do it manually or (better) with `godot --doctool` (with `godot` being the compiled Godot build from your PR changes).

File: doc/classes/RayCast3D.xml [hunk 2]
```diff
@@ -118,7 +118,7 @@
 			The custom color to use to draw the shape in the editor and at run-time if [b]Visible Collision Shapes[/b] is enabled in the [b]Debug[/b] menu. This color will be highlighted at run-time if the [RayCast3D] is colliding with something.
 			If set to [code]Color(0.0, 0.0, 0.0)[/code] (by default), the color set in [member ProjectSettings.debug/shapes/collision/shape_color] is used.
 		</member>
-		<member name="debug_shape_thickness" type="float" setter="set_debug_shape_thickness" getter="get_debug_shape_thickness" default="2.0">
+		<member name="debug_shape_thickness" type="int" setter="set_debug_shape_thickness" getter="get_debug_shape_thickness" default="2.0">
```
Comment (akien-mga) [inline doc/classes/RayCast3D.xml:121]: The default for an int becomes `2`.

File: scene/main/viewport.cpp (hunk 1 above)
Comment (akien-mga) [reply scene/main/viewport.cpp:3952]: Just this one left.

Comment (Ice-Cube69) [reply scene/main/viewport.cpp:3952]: Missed this message, working on it!

**Comment by pycbouh** [comment]
FYI there is something wrong with your branch, your last force push removed all commits and changes from it. There was probably a bad rebase or squash.

**Comment by Ice-Cube69** [comment]
Ah I see, i''ll make a new PR immidiately.

**Comment by Ice-Cube69** [comment]
Made a new PR #56667

**Pull request closed without merging**